src/algo.rs

```diff
@@ -151,18 +165,21 @@ impl Forest {
                 };
 
                 if width_compatible && height_compatible {
-                    return cache.result.clone();
+                    return Some(cache.result.clone());
                 }
 
                 if cache.node_size == node_size && cache.parent_size == parent_size {
-                    return cache.result.clone();
+                    return Some(cache.result.clone());
                 }
             }
         }
 
-        // Define some general constants we will need for the remainder
-        // of the algorithm.
+        None
```

Comment:
Not too versed in Rust yet so bear with me: Since all other branches in the code has `return Foo`, should this be `return None` instead of simply `None`?

I know they are functionally the same, but when it comes to readability, does it help to stick with one pattern instead of mixing two?

Reply:
I think clippy yells at you if you have an unnecessary return :')